packages/expo-modules-core/ios/Swift/Methods/ConcreteMethod.swift

```diff
@@ -0,0 +1,89 @@
+
+public struct ConcreteMethod<Args, ReturnType>: AnyMethod {
+  public typealias ClosureType = (Args) -> ReturnType
+
+  public let name: String
+
+  public var takesPromise: Bool {
+    return argTypes.last?.canCast(Promise.self) ?? false
+  }
+
+  public var argumentsCount: Int {
+    return argTypes.count - (takesPromise ? 1 : 0)
+  }
+
+  public var queue: DispatchQueue?
+
+  let closure: ClosureType
+
+  let argTypes: [AnyArgumentType]
+
+  init(
+    _ name: String,
+    argTypes: [AnyArgumentType],
+    queue: DispatchQueue? = nil,
+    _ closure: @escaping ClosureType) {
+    self.name = name
+    self.argTypes = argTypes
+    self.queue = queue
+    self.closure = closure
```

Comment:
In TS and Kotlin you can declare properties in the constructor signature what let's you omit this boilerplace assignings. Swift doesn't have such mechanism, has it?

Reply:
Nope, it doesn't have something like this yet. However, structs have default initializer so I probably can remove this init entirely now (it did have some other logic before so I had to explicitly define init)